.github/workflows/test.yaml

```diff
@@ -541,7 +572,8 @@ jobs:
       - narrativedocs
       - apidocs
       - static-checks
-      - release-publish
+      - release-build
+      - release-publish-success-or-skipped
```

Comment:
Do we need this extra job ?

To keep it simple, I was thinking that we can have `release-publish` as an "optional" steps.

We can keep only `release-build` here

It will be skipped on all bracnhes...and we can check that it is executed for a tag.

Reply:
We need this so if a publish fails it's detected by the all-successful step

Reply:
But we don't run this job for a PR ? 
The `all-success` job is designed to prevent merging a PR with failed jobs.


------

Also, the job is triggered only for a tag... so even if it fails. there is not much we can do, other than create a new tag ... or delete and recreate the tag